### PR TITLE
fix: register login on 401 and relax regex rules

### PR DESCRIPTION
**Pull request by Selchuk-Karakus**
The problem was when the registration was complete the eventual response was not been received on time for the login to do its user login authentication so every time it made an attempt the values it needed were not up to date. I used a setTimeout to execute the login after a certain amount of time has passed so it has the correct values to process.  And I have also relaxed some regex rules to give the user a bit more freedom when picking a password.

### CODE REVIEW

**Comment by Selchuk-Karakus** [comment]
The await keyword was missing. This was the root of the problem. We no longer need to hold up the program using setTimeout to wait for the necessary value to log in the user. We're able to do the same thing by using the await keyword so the value is settled before we ask for it. @mahri09 @HamraKhan

**Comment by mahri09** [comment]
the explanation was beneficial thank you!

File: client/src/services/validationService.js [hunk 1]
```diff
@@ -2,71 +2,84 @@
...
+  const result = /^[a-zA-Z0-9]{5,}/gm.test(password);
+  return result;
 };
 
 // Must contain at least 5 letters
 const isValidPasswordConf = (password) => {
-  return /^[a-zA-Z]{5,}/.test(password);
+  const result = /^[a-zA-Z0-9]{5,}/gm.test(password);
+  return result;
 };
```
Comment (mahri09) [inline client/src/services/validationService.js:66]: it must contain only 5 letters yeah no need number!?

File: routes/auth.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ router.post("/register", async function (req, res, next) {
     };
 
     //use create/insert method to insert object into postgresql database
-    insertMember(userToStore);
+    await insertMember(userToStore);
```
Comment (mahri09) [inline routes/auth.js:63]: we missed await in here yeah now I got it!

**Review by mahri09 — COMMENTED** [review]
I am happy you fixed it! well done Selchuk

File: client/src/services/validationService.js (hunk 1 above)
Comment (Selchuk-Karakus) [reply client/src/services/validationService.js:66]: A minimum of 5 characters is required. It must start with a letter. Can include numbers too.

File: routes/auth.js (hunk 2 above)
Comment (Selchuk-Karakus) [reply routes/auth.js:63]: Exactly! This is what was causing the problem, the missing await keyword.

**Pull request merged**